c/src/ml-api-common.c

```diff
@@ -1647,3 +1647,202 @@ ml_option_get (ml_option_h option, const char *key, void **value)
+  }
+
+  if (!value) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'value' is NULL. It should be a valid void**");
+  }
+
+  return _ml_info_get_value ((ml_info_s *) information, key, value);
+}
+
+/**
+ * @brief Internal function for destroy ml_info_list_s instance.
+ */
+static void
+_ml_info_list_destroy (ml_info_list_s * list)
+{
+  guint i;
+
+  for (i = 0; i < list->length; ++i) {
+    _ml_info_destroy (list->info[i]);
+  }
+
+  g_free (list->info);
+  g_free (list);
+}
+
+/**
+ * @brief Internal function for getting length of ml_info_list_s instance.
+ */
+static unsigned int
+_ml_info_list_length (ml_info_list_s * list)
+{
+  return list->length;
+}
+
+/**
+ * @brief Internal function for getting index-th ml_information_h instance in ml_info_list_s instance.
+ */
+static ml_information_h
+_ml_info_list_get (ml_info_list_s * list, unsigned int index)
+{
+  if (index >= list->length)
+    return NULL;
+
+  return (ml_information_h) list->info[index];
+}
+
+/**
+ * @brief Destroys the ml-information-list instance.
+ */
+int
+ml_information_list_destroy (ml_information_list_h list)
+{
+  check_feature_state (ML_FEATURE);
+
+  if (!list) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'list' is NULL. It should be a valid ml_information_list_h, which should be created by ml_information_list_create().");
+  }
+
+  _ml_info_list_destroy ((ml_info_list_s *) list);
+
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Gets the number of ml-information in ml-information-list instance.
+ */
+int
+ml_information_list_length (ml_information_list_h list, unsigned int *length)
+{
+  check_feature_state (ML_FEATURE);
+
+  if (!list) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'list' is NULL. It should be a valid ml_information_list_h, which should be created by ml_information_list_create().");
+  }
+
+  if (!length) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'length' is NULL. It should be a valid unsigned int*");
+  }
+
+  *length = _ml_info_list_length ((ml_info_list_s *) list);
+
+  return ML_ERROR_NONE;
+}
+
+/**
+ * @brief Gets a ml-information in ml-information-list instance with given index.
+ */
+int
+ml_information_list_get (ml_information_list_h list, unsigned int index,
+    ml_information_h * information)
+{
+  check_feature_state (ML_FEATURE);
+
+  if (!list) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'list' is NULL. It should be a valid ml_information_list_h, which should be created by ml_information_list_create().");
+  }
+
+  if (!information) {
+    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+        "The parameter, 'information' is NULL. It should be a valid ml_information_h*");
+  }
+
+  *information =
+      (ml_information_h *) _ml_info_list_get ((ml_info_list_s *) list, index);
```

Comment:
function return 0 on success. Otherwise a negative error value.
so, If index is out of range, the information will be null.
How about add
if (information == NULL)
     return "negative error value"
or check index at the beginning of the function
```
 if (index >= list->length)
    return NULL;
```

Reply:
Thanks I omitted checking the value of `_ml_info_list_get`. I updated the code PTAL

Reply:
WC^^